src/goRunTestCodelens.ts

```diff
@@ -95,7 +95,21 @@ export class GoRunTestCodeLensProvider extends GoBaseCodeLensProvider {
 				codelens.push(new CodeLens(func.location.range, runTestCmd));
 				codelens.push(new CodeLens(func.location.range, debugTestCmd));
 			});
-			return codelens;
 		});
+
+		getTestFunctions(document, hasBenchmarkFunctionPrefix).then(benchmarkFunctions => {
+			benchmarkFunctions.forEach(func => {
+				let runBenchmarkCmd: Command = {
+					title: 'run benchmark',
+					command: 'go.benchmark.cursor',
+					arguments: [ { functionName: func.name, isBenchmark: true } ]
+				};
+
+				codelens.push(new CodeLens(func.location.range, runBenchmarkCmd));
+			});
+
+		});
+
+		return Promise.resolve(codelens);
```

Comment:
This Promise should resolve only after both tests and benchmark functions are found